rita_client/src/exit_manager/mod.rs

```diff
@@ -590,11 +591,62 @@ fn initialize_selected_exit_list(exit: String, server: ExitServer) {
     });
 }
 
+/// Retrieve a unix timestamp from the exit's mesh IPv6
+pub async fn get_exit_time(exit: ExitServer) -> Option<i64> {
+    info!("Getting the exit time");
+    let exit_ip = exit.subnet.ip();
+    let exit_port = exit.registration_port;
+    let url = format!("http://[{}]:{}/time", exit_ip, exit_port);
+
+    let client = awc::Client::default();
+    let response = match client.get(&url).send().await {
+        Ok(mut response) => {
+            trace!("Response is {:?}", response.status());
+            trace!("Response is {:?}", response.headers());
+            response.json().await
+        }
+        Err(e) => {
+            error!("Failed to get exit time stamp {:?}", e);
+            return None;
+        }
+    };
+
+    let exit_time: ExitTimeStamp = match response {
+        Ok(a) => a,
+        Err(e) => {
+            error!("Failed to get exit time stamp {:?}", e);
+            return None;
+        }
+    };
+
+    Some(exit_time.secs_unix_epoch)
+}
+
+/// Set the router's time using "date -s \"@12345678\""
+pub async fn set_local_time(unix_time: i64) {
+    let t = format!("\"@{}\"", unix_time);
+    match KI.run_command("date", &["-s", t.as_str()]) {
+        Ok(_) => {}
+        Err(e) => error!("Failed to set local time {:?}", e),
+    }
+}
+
+/// Get time from the exit; if it's more than 60 secs different than local time, then set the local time to it  
+pub async fn maybe_set_local_to_exit_time(exit: ExitServer) {
```

Comment:
we should log when we set the time and maybe have the difference value be a constant? That way it can be more easily adjusted.

Reply:
made a constant for it at the top of the file.  is there a better place?  Will add an info log statement as well.